### PR TITLE
feat(provider-config): add http_proxy option to provider config

### LINKED ISSUE #139: provider specific proxy config option
Hi,

it would be nice a have a provider specific proxy config option. (for example like aws provider https://registry.terraform.io/providers/hashicorp/aws/latest/docs#http_proxy)

I know in the current version it's possible via env vars, but running in a large stack it effects every provider and can cause problems.

Do you think, there's any chance to implement this?

Thank you already in advance.

### PR DESCRIPTION
**Pull request by kaiehrhardt**
fixes #139

### CODE REVIEW

File: sonarqube/provider.go [hunk 1]
```diff
@@ -119,6 +123,13 @@ type ProviderConfiguration struct {
 
 func configureProvider(d *schema.ResourceData) (interface{}, error) {
 	transport := cleanhttp.DefaultPooledTransport()
+	if proxy, ok := d.Get("http_proxy").(string); ok {
```
Comment (jdamata) [inline sonarqube/provider.go:126]: Under the hood `cleanhttp.DefaultPooledTransport ` uses [https://pkg.go.dev/net/http#ProxyFromEnvironment](https://pkg.go.dev/net/http#ProxyFromEnvironment). 

This property is described as `ProxyFromEnvironment returns the URL of the proxy to use for a given request`

I still don't understand how this is functionally different from setting the proxy directly as SONAR_HOST but thats ok. If i find some time, i'll try write it out myself with an nginx proxy. 

I approved the workflow for the acceptance tests to run, if you can get the tests to run, im ok merging this in

Comment (kaiehrhardt) [reply sonarqube/provider.go:126]: Not every proxy is a reverse proxy forwarding requests to the corresponding webserver endpoint. Our usecase is more about routing traffic via a proxy connection to a jumphost in an internal aws vpc. So the proxy doesn't know where to route the traffic. That's done on client side.

> Under the hood `cleanhttp.DefaultPooledTransport ` uses https://pkg.go.dev/net/http#ProxyFromEnvironment.
> 
> This property is described as `ProxyFromEnvironment returns the URL of the proxy to use for a given request`
> 

I know. In large stacks, that call more then one provider, you want to specify which provider is using a proxy.

> I approved the workflow for the acceptance tests to run, if you can get the tests to run, im ok merging this in

Sounds good.

**Comment by kaiehrhardt** [comment]
Not sure, but it doesn't look like the problems are related to my changes. `make testacc` works.

**Comment by kaiehrhardt** [comment]
rebased

**Review by jdamata — APPROVED** [review]

**Pull request merged**